module_macaddress.psm1

```diff
@@ -0,0 +1,102 @@
+<#
+.SYNOPSIS
+    A simple script to retrieve MAC Address vendor by supplying MAC address
+.DESCRIPTION
+    This script downloads wireshark's Vendors Mac Address List from their gitweb.
+    This script has two functions in it:
+        * Update-MacAddressVendor
+        * Get-MacAddressVendor
+
+    Update-MacAddressVendor downloads the current vendor file and places it in a given directory, 
+    determined by the $DocumentPath variable.
+
+    Get-MacAddressVendor searches through the downloaded file for the three first octets of the MAC Address, 
+    and hopefully retrieves the correct vendor of it. 
+
+.EXAMPLE
+    # Update Vendor List
+    Update-MacAddressVendor
+
+    # Search through vendor list
+    Get-MacAddressVendor -MacAddress '00-50-56-C0-00-00'
+
+    ===================RESULT===================
+    00:50:56	Vmware	VMware, Inc.
+    ============================================
+    
+
+.NOTES
+    Script written by CodeBarbarian
+#>
+
+
+
+# Incase of a proxy
+[System.Net.WebRequest]::DefaultWebProxy.Credentials = [System.Net.CredentialCache]::DefaultCredentials
+
+
+################################# Script Config ##################################
+# -------------------------- Script Dependant Variables -------------------------#
+# Document Path - Where the vendor mac addresses textfile will be located
+$DocumentPath = ("c:$($ENV:HOMEPATH)\desktop")
+##################################################################################
+
+
+
+function Update-MacAddressVendor {
+    # Path to the wireshark repository
+    $APIUrl = 'https://code.wireshark.org/review/gitweb?p=wireshark.git;a=blob_plain;f=manuf;hb=HEAD'
+
+    $WebClient = New-Object System.Net.WebClient
+    $WebClient.DownloadFile($APIUrl, "$($DocumentPath)\vendor_macaddresses.txt")
+
+    Write-Host ("Update of MAC vendor list complete.") -ForegroundColor Green

```

Comment:
Please use `Write-Verbose` instead (see e.g. [this](http://www.jsnover.com/blog/2013/12/07/write-host-considered-harmful/) blog post from the PowerShell inventor himself about why `Write-Host` should not be used). You will then also need to add the line `[CmdletBinding()]Param()` below the function declaration to enable automatic propagation of `$VerbosePreference`

Reply:
Thanks, so you will always need the '[cmdletbinding()]' and 'param()' in any given function to allow for automatic propagation?

Reply:
Yes. The `[cmdletbinding()]` attribute makes it an advanced function and only advanced functions allow you to pass the `$VerbosePreference` in either explicitly (using `-Verbose`) or implicitly when it is being called from another function (this is the case of automatic propagation). Have a read about 'advanced functions', they give so much more stuff for free (such as e.g. the validation attributes that I mention elsewhere and also making parameters mandatory, etc.). For an example, just press `Ctrl+J` in the ISE (which gives you a list of snippets) and choose `Cmdlet (advanced function) - complete` and you will see a long examples with a lot of example use cases that advanced functions allow you to do.